LayoutTests/accessibility/element-reflection-ariacontrols.html

```diff
@@ -0,0 +1,55 @@
+<!DOCTYPE html>
+<script src="../resources/js-test.js"></script>
+<script src="../resources/accessibility-helper.js"></script>
+<div id="panel1">First panel</div>
+<div id="tab1" role="tab">Tab 1</div>
+<div id="wrapper" tabindex="0">
+  <div class="panel">Second panel</label>
+</div>
+<div id="tab2" role="tab">Tab 2</div>
+<div id="panel3">Third panel</div>
+<x-panel></x-panel>
+
+<script>
+  class XPanel extends HTMLElement {
+    constructor() {
+      super();
+      this.attachShadow({ mode: "open" });
+      let tab = document.createElement("div");
+      tab.id = "innertab";
+      tab.role = "tab";
+      tab.textContent = "Tab 3";
+      tab.ariaControlsElements = [panel3];
```

Comment:
This isn't really testing shadow DOM. We need to put the whole custom element into a shadow tree as I'm doing in https://github.com/WebKit/WebKit/pull/4561

Reply:
This is failing without this patch and fixed by this patch. It's referring an element outside the shadow tree from it.

I can put the whole thing in a shadow tree like in the your tests, but would that be something on top of this or a replacement? This is testing something that is currently broken, not sure what the new test will give us on top of this.

Reply:
Interesting. If this test case is also failing then we need to add it as another test case.

Reply:
Sorry because my last comment here was misleading. "This is failing without this patch and fixed by this patch", as I mentioned on a previous comment I was having problems testing the Shadow DOM related parts of my tests, so this was not actually fixed by my patch.

The problems were happening on `run-webkit-tests` but not with `run-minibrowser` where I could see the relationships properly set on the a11y info in the inspector or using a screen reader with my patch (that's why I said "fixed by this patch").

The good thing is that those problems are fixed by #4561, this case and the one where the whole custom element is in the Shadow Tree would work the same. Not sure if we need 2 separated test cases or 1 is enough.

Reply:
I have added more test cases, some modifying  the `id` and still keeping the reference, and others moving an element out of a shadow tree and still keeping the reference working.

Please let me know if you think we need to test more cases, for now I'm marking this conversation as resolved.